Multisig/UI/Transactions/TransactionsViewModel.swift

```diff
@@ -0,0 +1,428 @@
+//
+//  TransactionListViewModel.swift
+//  Multisig
+//
+//  Created by Moaaz on 5/27/20.
+//  Copyright © 2020 Gnosis Ltd. All rights reserved.
+//
+
+import Foundation
+import Combine
+import SwiftCryptoTokenFormatter
+import BigInt
+
+struct TransactionsList {
+    struct Section {
+        var name: String
+        var transactions: [BaseTransactionViewModel]
+
+        var isEmpty: Bool {
+            transactions.isEmpty
+        }
+    }
+
+    var sections: [Section] = []
+
+    var isEmpty: Bool {
+        sections.allSatisfy { $0.isEmpty }
+    }
+}
+
+class BaseTransactionViewModel {
+
+    var nonce: String?
+    var status: TransactionStatus
+    var date: Date?
+    var formattedDate: String
+    var confirmationCount: Int?
+    var threshold: Int?
+
+    init() {
+        status = .success
+        formattedDate = ""
+        date = Date()
+    }
+}
+
+
+class TransferTransaction: BaseTransactionViewModel {
+
+    var address: String
+    var isOutgoing: Bool
+    var amount: String
+    var tokenSymbol: String
+
+    override init() {
+        amount = "0"
+        tokenSymbol = "ETH"
+        isOutgoing = true
+        address = ""
+        super.init()
+    }
+}
+
+
+class ChangeMasterCopyTransaction: BaseTransactionViewModel {
+
+    var contractVersion: String
+    var contractAddress: String
+
+    override init() {
+        contractVersion = ""
+        contractAddress = ""
+        super.init()
+    }
+}
+
+
+class SettingChangeTransaction: BaseTransactionViewModel {
+
+    var title: String
+
+    override init() {
+        title = ""
+    }
+
+}
+
+class CustomTransaction: TransferTransaction {
+    var dataLength: Int
+
+    override init() {
+        dataLength  = 98
+        super.init()
+    }
+}
+
+extension TransactionStatus {
+    static let queueStatuses = [TransactionStatus.pending, .waitingConfirmation, .waitingExecution]
+    static let historyStatuses = [TransactionStatus.success, .failed, .canceled]
+
+    var isInQueue: Bool {
+        Self.queueStatuses.contains(self)
+    }
+
+    var isInHistory: Bool {
+        Self.historyStatuses.contains(self)
+    }
+}
+
+
+class TransactionsViewModel: ObservableObject {
+    @Published var transactionsList = TransactionsList()
+
+    @Published var isLoading: Bool = true
+    @Published var errorMessage: String? = nil
+
+
+    private let safe: Safe
+    private var subscribers = Set<AnyCancellable>()
+
+    init(safe: Safe) {
+        self.safe = safe
+        loadData()
+    }
+
+    func loadData() {
+        isLoading = true
+        Just(safe.address!)
+            .setFailureType(to: Error.self)
+            .flatMap { address in
+                Future<TransactionsList, Error> { promise in
+                    DispatchQueue.global().async {
+                        do {
+                            let info = try App.shared.safeTransactionService.safeInfo(at: address)
```

Comment:
Is it a temporary solution? I thought that we should have all the required info from the endpoint.

Reply:
There is open ticket for that, but it seems we have to live with this way for the time being